source/opt/instrument_pass.cpp

```diff
@@ -213,14 +213,14 @@ uint32_t InstrumentPass::GenStageInfo(uint32_t stage_idx,
       load_id = GenVarLoad(
           context()->GetBuiltinInputVarId(uint32_t(spv::BuiltIn::InvocationId)),
           builder);
-      ids[2] = load_id;
+      ids[2] = GenUintCastCode(load_id, builder);
     } break;
     case spv::ExecutionModel::TessellationControl: {
       // Load and store InvocationId and PrimitiveId
       uint32_t load_id = GenVarLoad(
           context()->GetBuiltinInputVarId(uint32_t(spv::BuiltIn::InvocationId)),
           builder);
-      ids[1] = load_id;
+      ids[1] = GenUintCastCode(load_id, builder);
```

Comment:
I'm confused why this only done for `InvocationID`, `gl_PrimitiveID` is also a `int` in glslang

Vulkan just cares it is 32-bit, not signed/unsigned
![image](https://github.com/KhronosGroup/SPIRV-Tools/assets/115671160/727d4dcd-9348-42ae-a7ec-298b420bd3bb)

since all we do is `OpLoad` and pass it around until it gets `OpStore` in the output buffer (aka, we don't do any arithmetic operations on it) I almost think it might be easier to treat things as `uint` everywhere in SPIR-V and in VVL when we read in the buffer, for stages like Tessellation we cast it there

Reply:
I realize changing this here is local to just the instrumentation pass, but the core issue is `IRContext::GetBuiltinInputVarId` is marking this a `uint32`

Reply:
gl_PrimitiveID  already becomes a uint somehow. I have no clue why.

Reply:
> since all we do is `OpLoad` and pass it around until it gets `OpStore` in the output buffer (aka, we don't do any arithmetic operations on it) I almost think it might be easier to treat things as `uint` everywhere in SPIR-V and in VVL when we read in the buffer, for stages like Tessellation we cast it there

Do you mean GenUintCastCode() all the things?

Reply:
There is nothing stopping a HLL from setting the `InvocationID` to be either a `uint` or `int` so we would need to guard against that either way

I guess I am still not sure where in practice this becomes a problem, I can't believe it is actually valid to have a negative `InvocationID` or anyone is going to have a value more then 2^31

So I assume the issue is some HLL has set this to be a `int` and we need it passed in as a `uint` for the function and you are getting spirv-val error that the function argument types are not matching?